internal/provider/resource_access_list.go

```diff
@@ -124,17 +123,31 @@ func resourceAccessListDelete(ctx context.Context, d *schema.ResourceData, meta
 		return diag.FromErr(err)
 	}
 
-	aResp := *accessList.Addresses
-	addressesQP := astra.AddressesQueryParam{}
-	for _, v := range aResp {
-		address:= *v.Address
-		addressesQP = append(addressesQP, address)
-	}
+	// aResp := *accessList.Addresses
+	// addressesQP := astra.AddressesQueryParam{}
+	// for _, v := range aResp {
+	// 	address:= *v.Address
+	// 	addressesQP = append(addressesQP, address)
+	// }
+
+	// params :=  &astra.DeleteAddressesOrAccessListForDatabaseParams{
+	// 	&addressesQP,
+	// }
+	// client.DeleteAddressesOrAccessListForDatabase(ctx, astra.DatabaseIdParam(databaseID), params)
 
-	params :=  &astra.DeleteAddressesOrAccessListForDatabaseParams{
-		&addressesQP,
+	// The above code should work, but Astra seems to only delete the first address passed as a query param
+	// Until it's fixed in Astra, call DELETE for each address
+	aResp := *accessList.Addresses
+	if len(aResp) > 0 {
```

Comment:
Formatting looks to be off a bit, maybe run `go fmt`.

Reply:
I did run it, but it didn't change anything. Opening this in a different editor shows me that the spacing is all messed up. I'll update that.

Reply:
I take it back, the tabs/spacing is the same as the rest. It looks funny for the bit of code that is commented out, but it's the same for the code changes.